src/Utilities/functions.jl

```diff
@@ -331,11 +308,14 @@ function modifyfunction(f::MOI.ScalarQuadraticFunction, change::MOI.ScalarCoeffi
                                 f.quadratic_terms, f.constant)
 
 end
+if VERSION >= v"0.7.0-DEV.2813"
+    using SparseArrays: sparsevec
+end
 function _modifycoefficients(n, terms::Vector{<:MOI.VectorAffineTerm}, variable::MOI.VariableIndex, new_coefficients)
     terms = copy(terms)
     # Maps between rows in the `VectorAffineTerm`s and indices in new_coefficients
-    rowmap = zeros(Int, n)
-    rowmap[map(c -> c[1], new_coefficients)] = 1:length(new_coefficients)
+    # Uses sparse vector since `new_coefficients` also uses a sparse representation 
+    rowmap = sparsevec(getindex.(new_coefficients, 1), 1:length(new_coefficients), n)
```

Comment:
The advantage of the approach currently used is the O(1) access to the new coefficient. With a sparse vector, it is O(log(n)) (if it does sorted search which I guess it does). It is also annoying to add a dependency on `SparseArrays` since there are discussion on removing it from Base.
We might use a `Dict` here which offers O(1) access and memory proportional to the number of elements in `new_coefficients`

Reply:
Good point, didn't think of that. I will change it to `Dict`.